test/search/query-variants.test.ts

```diff
@@ -0,0 +1,48 @@
+import {queryVariants} from '../../src/search/query-variants';
+
+describe('Search query', () => {
+  it('transforms simple query', () => {
+    expect(queryVariants('test')).toEqual(
+      new Map([
+        ['?query', 'test'],
+        ['?booleanQuery', "'test'"],
+      ])
+    );
+  });
+
+  it('transforms multiple word query', () => {
+    expect(queryVariants('Dr. H. Colijnstraat')).toEqual(
+      new Map([
+        ['?query', 'dr. h. colijnstraat'],
+        ['?booleanQuery', "'dr.' AND 'h.' AND 'colijnstraat'"],
+      ])
+    );
+  });
+
+  it('trims whitespaces', () => {
+    expect(queryVariants('   a   b  c  ')).toEqual(
+      new Map([
+        ['?query', 'a b c'],
+        ['?booleanQuery', "'a' AND 'b' AND 'c'"],
+      ])
+    );
+  });
+
+  it('skips already present boolean operators', () => {
+    expect(queryVariants('a AND b c or d')).toEqual(
+      new Map([
+        ['?query', 'a and b c or d'],
+        ['?booleanQuery', "'a' and 'b' AND 'c' or 'd'"],
+      ])
+    );
+  });
+
+  it('escapes quotation marks', () => {
+    expect(queryVariants("Rex Stewart's Big Eight")).toEqual(
+      new Map([
+        ['?query', "rex stewart's big eight"],
+        ['?booleanQuery', "'rex' AND 'stewart\\'s' AND 'big' AND 'eight'"],
```

Comment:
Check: does SPARQL need double `\\` for escaping chars?

Reply:
To be honest: I'm not sure about SPARQL. However, I do know that Virtuoso needs the double `\\` for escaping. See for example this NTA query: https://api.triplydb.com/s/oJd_r6Er5. The crucial part is `"'D\\'hondt'"`. If you omit one or two of the `\` before `'` then Virtuoso throws an error ("Unterminated single-quoted word or phrase").

Reply:
Alternatively, we could use double quotes rather than single quotes. See for example this NTA query: https://api.triplydb.com/s/CuAAHslLZ, part `'"D\'hondt"'`.

Reply:
Come to think of it: it could be that current (advanced) NoT users already provide the NoT with queries in which both quote types are being used, e.g. `"'D\\'hondt'"` or  `'"D\'hondt"'`. Should we add a check for this (similar to the check if boolean operators are already present), so as to prevent duplicate escaping (e.g. `'D\\'Hondt'` becomes `\\'D\\\\'Hondt\\'`)?

Reply:
> To be honest: I'm not sure about SPARQL. However, I do know that Virtuoso needs the double `\\` for escaping. 

Good point: we’re talking about Virtuoso- rather than SPARQL-specifics here.

> Should we add a check for this (similar to the check if boolean operators are already present), so as to prevent duplicate escaping (e.g. `'D\\'Hondt'` becomes `\\'D\\\\'Hondt\\'`)?

Yes. I’ll change the code to:

1. prevent duplicate scanning by only escape not-yet-escaped single quotes (so `'` becomes `\\'` but `\\'` does not become `\\\\');
2. only quote not-yet-quoted words (so `hond` becomes `'hond'` but `'hond'` stays `'hond'` and `"hond"` stays `"hond"`).

Reply:
@sdevalk will check how double quotation marks `"` behave in Virtuoso. Do we have to escape them too?

Reply:
Is this still relevant now that I’ve added a `RAW` query type here?